File: 2018-12-23/8958/8958.java
import java.io.*;

class Main{
	public static void  main(String[] args) throws Exception{
		BufferedReader bfr = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(bfr.readLine());
		String OX;
		int last = 0;
		int []result = new int[1000];
		for(int i = 0; i < N; i++){
			result[i]=0;
			last=0;
			OX = bfr.readLine();
			for (int j = 0; j < OX.length(); j++){
				if(OX.charAt(j) == 'X')
					last=0;
				else {
					last+=1;
					result[i]+=last;
				}
			}
		}
		for(int i = 0;i < N; i++)
			System.out.println(result[i]);
	}
}

File: 2018-12-23/8958/8958.py
# import sys

# N = int(sys.stdin.readline())

# result = [0 for i in range(0, 100)]

# for i in range(0, N):
# 	last = 0
# 	OX = sys.stdin.readline()
# 	for j in range(0, len(OX)-1):
# 		if(OX[j]=="X"):
# 			last=0
# 		else:
# 			last+=1
# 			result[i] += last	

# for i in range(0,N):
# 	print(result[i])

#1등 코드

from sys import stdin
for _ in range(int(stdin.readline())):
    print(sum(sum(range(1, len(i+1))) for i in stdin.readline().strip().split("X")))	